docker/puppetdb/docker-entrypoint.d/10-configure-ssl.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+master_running() {
+    status=$(curl --silent --fail --insecure "https://${PUPPETSERVER_HOSTNAME}:8140/status/v1/simple")
+    test "$status" = "running"
+}
+
+PUPPETSERVER_HOSTNAME="${PUPPETSERVER_HOSTNAME:-puppet}"
+
+if [ ! -f "/etc/puppetlabs/puppet/ssl/certs/${HOSTNAME}.pem" ] && [ "$USE_PUPPETSERVER" = true ]; then
+  # if this is our first run, run puppet agent to get certs in place
+  while ! master_running; do
+    sleep 1
+  done
+  set -e
+  /ssl.sh
+fi
+if [ ! -d "/etc/puppetlabs/puppetdb/ssl" ] && [ "$USE_PUPPETSERVER" = true ]; then
+  /ssl-setup.sh -f
+fi
```

Comment:
I initially tried breaking this script up into 3 distinct parts:
* 10-wait-for-puppetserver.sh (just does the curl+sleep)
* 20-get-ssl-cert.sh (calls /ssl.sh)
* 30-setup-ssl.sh (calls /ssl-setup.sh -f)

But that ended up ballooning out too much I thought, and considering how tightly coupled these steps are I ended up leaving them as one.

At that point this PR was a little less desirable, as I wasn't really decomposing anything.

But, we'll likely have a new script for analytics that we'll want to drop in here to run first, something like `00-analytics.sh` or whatever.  It'll be nice to have this pattern in place for that work, and just in general to keep the entrypoint logic decoupled from one another.